src/persist-client/src/internal/machine.rs

```diff
@@ -882,6 +813,104 @@ where
     }
 }
 
+impl<K, V, T, D> Machine<K, V, T, D>
+where
+    K: Debug + Codec,
+    V: Debug + Codec,
+    T: Timestamp + Lattice + Codec64,
+    D: Semigroup + Codec64 + Send + Sync,
+{
+    pub async fn start_reader_heartbeat_task(
```

Comment:
I didn't verify that these are exactly what the they were pre-move (could maybe be an issue if you created this by literally reverting the revert and they've changed in the meantime). Just want to verify that you did?

Reply:
+1 on verifying this!

Reply:
I attempted to, but I will double-check -- it is annoyingly hard to review after the squash.

Reply:
```diff
-    pub async fn start_reader_heartbeat_task(
-        self,
-        reader_id: LeasedReaderId,
-        gc: GarbageCollector<K, V, T, D>,
-    ) -> JoinHandle<()> {
+    pub async fn start_reader_heartbeat_task(self, reader_id: LeasedReaderId) -> JoinHandle<()> {
         let mut machine = self;
         spawn(|| "persist::heartbeat_read", async move {
             let sleep_duration = machine.applier.cfg.reader_lease_duration / 2;
@@ -843,10 +839,9 @@ where
                 }
 
                 let before_heartbeat = Instant::now();
-                let (_seqno, existed, maintenance) = machine
+                let (_seqno, existed, _maintenance) = machine
                     .heartbeat_leased_reader(&reader_id, (machine.applier.cfg.now)())
                     .await;
-                maintenance.start_performing(&machine, &gc);
 
                 let elapsed_since_heartbeat = before_heartbeat.elapsed();
                 if elapsed_since_heartbeat > Duration::from_secs(60) {
```

...and similar for the writer.